kernel/userspace.c

```diff
@@ -381,32 +383,46 @@ static void wordlist_cb(struct _k_object *ko, void *ctx_ptr)
 
 void _thread_perms_inherit(struct k_thread *parent, struct k_thread *child)
 {
-	struct perm_ctx ctx = {
-		thread_index_get(parent),
-		thread_index_get(child),
-		parent
-	};
+	bool parent_id_status;
+	bool child_id_status;
+	u32_t parent_id = 0U;
+	u32_t child_id = 0U;
+
+	parent_id_status = thread_index_get(parent, &parent_id);
+	child_id_status = thread_index_get(child, &child_id);
+
+	if (parent_id_status && child_id_status) {
+		struct perm_ctx ctx = {
+			.parent_id = parent_id,
+			.child_id = child_id,
+			.parent = parent
+		};
 
-	if ((ctx.parent_id != -1) && (ctx.child_id != -1)) {
 		_k_object_wordlist_foreach(wordlist_cb, &ctx);
 	}
 }
 
 void _thread_perms_set(struct _k_object *ko, struct k_thread *thread)
 {
-	int index = thread_index_get(thread);
+	u32_t index;
+	bool status;
+
+	status = thread_index_get(thread, &index);
 
-	if (index != -1) {
+	if (status) {
```

Comment:
Consistency: `status == true`? as seen earlier in this commit.

Reply:
if the variable inside "if" is a bool then it doenst need a check with "true". The previous use was with rb_contains which is incorrect. will change that.